tests/phpunit/tests/html-api/wpHtmlProcessorComments.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.7.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_WpHtmlProcessorComments extends WP_UnitTestCase {
+	/**
+	 * Ensures that different types of comments are processed correctly.
+	 *
+	 * @ticket 61530
+	 *
+	 * @dataProvider data_comments
+	 */
+	public function test_comment_processing( string $html, int $token_position, string $expected_token_name, ?string $expected_comment_type, string $expected_modifiable_text = '', string $expected_tag = null ) {
```

Comment:
It feels a bit awkward to have the order of the final three be:

nullable, default, string who default is null

It also looks like there is only one case where `$expected_modifiable_text` is set but `$expected_comment_type` is not. For consistency, I wonder if it would make sense to swap those two and make $expected_comment_type default to null?